### PR TITLE
Git Actions Workflow and Poetry

### PR DESCRIPTION
**Pull request by burgess01**
# Git Actions Workflow and Poetry

## Description

Merge Workflow and Poetry into main to restore the Git Flow Workflow. In Git Flow there is not a development branch, so we are merging this into main and then deleting the develop branch. This pull request will set up Github Actions and install the poetry suite for the other groups' use.

## Linked Issues

closes: #19 

## Type of Change

- [x] Feature
- [ ] Bug fix
- [ ] Documentation

## Contributors

- @ullrichd21 @laurennevill @burgess01 

## Reminder

 - All GitHub Actions should be in a passing state before any pull request is merged.
 - All PRs must be reviewed by at least one team member and one member of the Integration team!
 - Any issues this PR closes are tagged in the description!

### CODE REVIEW

**Review by ullrichd21 — APPROVED** [review]
This looks good.

*Note: We're merging this without passing tests due to the code coverage which doesn't exist.*

**Review by noorbuchi — APPROVED** [review]

**Review by watson02 — COMMENTED** [review]
Looks good so far

**Review by laurennevill — APPROVED** [review]

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  publish:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Set up Python 3.9
+        uses: actions/setup-python@v2
+        with:
+          python-version: '3.9'
```
Comment (mariakimheinert) [inline .github/workflows/publish.yml:17]: The project should also be built with the oldest supported Python version--3.7. Though using a newer version for building may not cause an issue, using the oldest version will better ensure that the release is compatible with all supported versions.

**Review by mariakimheinert — CHANGES_REQUESTED** [review]
Thanks for making these additions, Integration team! I have provided a few points of feedback below. Please let me know if you have any questions!

**Review by Michionlion — DISMISSED** [review]
This looks good to me, given that the issues raised by @mariakimheinert are addressed!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    name: Lint Code
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: Setup Python 3.9
+        uses: actions/setup-python@v2
+        id: setup-python
+        with:
+          python-version: 3.9
```
[the remark this answers is not in the document]
Comment (burgess01) [reply .github/workflows/main.yml]: I corrected this section to reflect the correct python version, thank you professor!

File: .github/workflows/publish.yml (hunk 1 above)
Comment (burgess01) [reply .github/workflows/publish.yml]: I corrected this to display the right version of python! I'll keep an eye out for things like this in the future.

**Review by ullrichd21 — APPROVED** [review]
I think all issues have been fixed, we should be ready to merge once all final reviews are done.

**Review by noorbuchi — APPROVED** [review]

**Review by watson02 — APPROVED** [review]
Looks good.

**Review by laurennevill — APPROVED** [review]
Looks good

**Review by Michionlion — APPROVED** [review]
Given that the requested changes have all been completed, I think this PR is feature-complete!

**Pull request merged**